Apps/Examples/Examples/All Examples/NavigationSimulatorExample.swift

```diff
@@ -0,0 +1,274 @@
+        CLLocationCoordinate2D(latitude: 45.49805, longitude: -122.68923),
+        CLLocationCoordinate2D(latitude: 45.49857, longitude: -122.68926),
+        CLLocationCoordinate2D(latitude: 45.49911, longitude: -122.68814),
+        CLLocationCoordinate2D(latitude: 45.49921, longitude: -122.68865),
+        CLLocationCoordinate2D(latitude: 45.49905, longitude: -122.6897),
+        CLLocationCoordinate2D(latitude: 45.49917, longitude: -122.69346),
+        CLLocationCoordinate2D(latitude: 45.49902, longitude: -122.69404),
+        CLLocationCoordinate2D(latitude: 45.49796, longitude: -122.69438),
+        CLLocationCoordinate2D(latitude: 45.49697, longitude: -122.69504),
+        CLLocationCoordinate2D(latitude: 45.49661, longitude: -122.69624),
+        CLLocationCoordinate2D(latitude: 45.4955, longitude: -122.69781),
+        CLLocationCoordinate2D(latitude: 45.49517, longitude: -122.69803),
+        CLLocationCoordinate2D(latitude: 45.49508, longitude: -122.69711),
+        CLLocationCoordinate2D(latitude: 45.4948, longitude: -122.69688),
+        CLLocationCoordinate2D(latitude: 45.49368, longitude: -122.69744),
+        CLLocationCoordinate2D(latitude: 45.49311, longitude: -122.69702),
+        CLLocationCoordinate2D(latitude: 45.49294, longitude: -122.69665),
+        CLLocationCoordinate2D(latitude: 45.49212, longitude: -122.69788),
+        CLLocationCoordinate2D(latitude: 45.49264, longitude: -122.69771),
+        CLLocationCoordinate2D(latitude: 45.49332, longitude: -122.69835),
+        CLLocationCoordinate2D(latitude: 45.49334, longitude: -122.7007),
+        CLLocationCoordinate2D(latitude: 45.49358, longitude: -122.70167),
+        CLLocationCoordinate2D(latitude: 45.49401, longitude: -122.70215),
+        CLLocationCoordinate2D(latitude: 45.49439, longitude: -122.70229),
+        CLLocationCoordinate2D(latitude: 45.49566, longitude: -122.70185),
+        CLLocationCoordinate2D(latitude: 45.49635, longitude: -122.70215),
+        CLLocationCoordinate2D(latitude: 45.49674, longitude: -122.70346),
+        CLLocationCoordinate2D(latitude: 45.49758, longitude: -122.70517),
+        CLLocationCoordinate2D(latitude: 45.49736, longitude: -122.70614),
+        CLLocationCoordinate2D(latitude: 45.49736, longitude: -122.70663),
+        CLLocationCoordinate2D(latitude: 45.49767, longitude: -122.70807),
+        CLLocationCoordinate2D(latitude: 45.49798, longitude: -122.70807),
+        CLLocationCoordinate2D(latitude: 45.49798, longitude: -122.70717),
+        CLLocationCoordinate2D(latitude: 45.4984, longitude: -122.70713),
+        CLLocationCoordinate2D(latitude: 45.49893, longitude: -122.70774)
+    ])
+}
+
+extension NavigationSimulatorExample: LocationConsumer {
+
+    func locationUpdate(newLocation: Location) {
+        let progress = navigationSimulator.distanceTravelled / navigationSimulator.routeLength
+        routeLine.lineTrimOffset = .constant([0, progress])
+        casingLine.lineTrimOffset = .constant([0, progress])
```

Comment:
I think the issue is that the routeLine and casingLine in iOS is a struct and once the property is updated, you need to set the line back to the style again.

One option is to use the low level style API, to set the lineTrimOffset to the style directly, with the desired layer id.